evm/vm/evm.py

```diff
@@ -125,10 +106,10 @@ def get_vm_class_for_block_number(cls, block_number):
         """
         Return the vm class for the given block number.
         """
-        range = find_range(tuple(cls.vms_by_range.keys()), block_number)
-        base_vm_class = cls.vms_by_range[range]
-        vm_class = base_vm_class.configure(db=cls.db)
```

Comment:
I believe that this line still needs to be preserved.  The `db` property is set on the class and there are some `classmethod`s on the `VM` class that either currently require or are likely to require db access.

This manual API of passing the DB down through the object classes isn't ideal as it's easy to forget to do so if you have better ideas I'm all ears.